### ISSUE TITLE
splitter: add an option to specify split sizes as fractions of the dataset; add validation fraction

### ISSUE THREAD

**Issue by youph** [data61/stellar-evaluation-plugins#28]
Currently the number of samples in train and test subsets is specified as counts, which can be inconvenient. We need an option to specify their sizes as fractions of the total.

Also, currently whatever does not go into training or test set, ends up in the validation set. This can lead to a huge validation set, or an empty validation set. We need to be able to avoid these extremes.
